src/scenic/syntax/compiler.py

```diff
@@ -253,6 +274,94 @@ def visit_Require(self, node: s.Require):
             )
         )
 
+    def visit_Take(self, node: s.Take):
+        action = ast.Tuple(self.visit(node.elts), loadCtx)
+        return self.generateInvocation(node, ast.copy_location(action, node))
+
+    def visit_Wait(self, node: s.Wait):
+        return self.generateInvocation(node, ast.Constant(()))
+
+    def visit_Terminate(self, node: s.Terminate):
+        termination = ast.Call(
+            ast.Name("makeTerminationAction", loadCtx), [ast.Constant(node.lineno)], []
+        )
+        return self.generateInvocation(node, termination)
+
+    def visit_Do(self, node: s.Do):
+        subHandler = ast.Attribute(
+            ast.Name(behaviorArgName, loadCtx), "_invokeSubBehavior", loadCtx
+        )
+        # TODO(shun): Check node has no more than one element inside behavior/monitors
```

Comment:
`do` can only take one argument inside behaviors and monitors. Need to check that `len(node.elts) == 1` if `self.inCompose` being added in #77 is `True`.